src/Providers/TOTPProvider.php

```diff
@@ -22,36 +15,18 @@ class TOTPProvider extends BootstrapMFAProvider implements MFAProvider
 {
     /**
      * @param string $token
-     * @param null $result
-     * @return bool|Member
-     * @throws ValidationException
-     * @throws PasswordEncryptor_NotFoundException
+     * @return bool|TOTP
      * @throws \Exception
      */
-    public function verifyToken($token, &$result = null)
+    public function fetchToken($token)
     {
-        if (!$result) {
-            $result = Injector::inst()->get(ValidationResult::class);
-        }
         $member = $this->getMember();
         if ($member && $member->ID) {
-            if (!$token) {
-                $result->addError('Invalid or missing second factor token');
-            } else {
-                $secret = Base32::decode($member->TOTPSecret);
-                $algorithm = TOTPAuthenticator::get_algorithm();
+            $algorithm = TOTPAuthenticator::get_algorithm();
 
-                $totp = new Totp($algorithm);
-                $key = $totp->GenerateToken($secret);
-                $user_submitted_key = $token;
-                if ($user_submitted_key !== $key) {
-                    $result->addError('Invalid or missing second factor token');
-                } else {
-                    return $this->member;
-                }
-            }
+            return TOTP::create($member->TOTPSecret, 30, $algorithm);
```

Comment:
So the second argument is a "period" (https://github.com/Spomky-Labs/otphp/blob/v9.0/src/TOTP.php#L29) ? Should this be configurable? At the moment it looks a little like a magic number

Reply:
No stress, I suspect we'll probably have a whip over at a later stage and swap things like this out in bulk so I'll leave it until that time comes (if it does)